test_unstructured_ingest/test-ingest-azure-cognitive-search.sh

```diff
@@ -65,16 +67,27 @@ PYTHONPATH=. ./unstructured/ingest/main.py \
   --endpoint "$AZURE_SEARCH_ENDPOINT" \
   --index "$DESTINATION_INDEX"
 
-echo "sleeping 5 seconds to let index finish catching up after writes"
-sleep 5
 
-# Check the contents of the index
-docs_count=$(curl "https://utic-test-ingest-fixtures.search.windows.net/indexes/$DESTINATION_INDEX/docs/\$count?api-version=$API_VERSION" \
-  --header "api-key: $AZURE_SEARCH_API_KEY" \
-  --header 'content-type: application/json' | jq)
+
+docs_count=0
+attempt=1
+while [ "$docs_count" -eq 0 ] && [ "$attempt" -lt 6 ]; do
+  echo "attempt $attempt: sleeping 10 seconds to let index finish catching up after writes"
+  sleep 10
+
+  # Check the contents of the index
+  docs_count=$(curl "https://utic-test-ingest-fixtures.search.windows.net/indexes/$DESTINATION_INDEX/docs/\$count?api-version=$API_VERSION" \
+    --header "api-key: $AZURE_SEARCH_API_KEY" \
+    --header 'content-type: application/json' | jq)
+
+  echo "docs count pulled from Azure: $docs_count"
+
+  attempt=$((attempt+1))
+done
+
 
 expected_docs_count=0
-for i in $(jq length "$OUTPUT_DIR"/*); do
+for i in $(jq length "$OUTPUT_DIR"/**/*.json); do
   expected_docs_count=$((expected_docs_count+i));
```

Comment:
maybe we update the names of "docs_count" vs "expected_docs_count"?... since we have a lot going on here maybe something to make a name to make it clear that docs_count are the local written results and expected_docs_count is what was written to the destination?